packages/client3/components/design-system/button/button.tsx

```diff
@@ -1,163 +1,94 @@
-      {spinnerOnRight ? (
-        <Spinner size="sm" className={clsx(children ? "-my-2 -mr-1" : null)} />
-      ) : iconRight ? (
-        <Icon
-          name={iconRight}
-          size="sm"
-          className={clsx(children ? "-my-2 -mr-1" : null)}
-        />
-      ) : null}
-    </Component>
-  );
-});
-
-interface IconButtonProps
-  extends Omit<ButtonProps, "children" | "iconLeft" | "iconRight"> {
-  icon: IconProps["name"];
+export type ButtonProps = ButtonHTMLAttributes<HTMLButtonElement> & {
   /**
-   * Accessibility label. Must be provided since this form of button has no visible label.
+   * Content within the button
    */
-  label: string;
-}
+  children?: ReactNode;
+  disabled?: boolean;
+  buttonType?: "primary" | "secondary" | "tertiary";
+  iconLeft?: IconProps["name"];
+  iconRight?: IconProps["name"];
+  isLoading?: boolean;
+};
 
-export const IconButton = forwardRef<HTMLButtonElement, IconButtonProps>(
-  function IconButton({ icon, size = "md", className, ...rest }, ref) {
+export const Button = forwardRef<HTMLButtonElement, ButtonProps>(
+  function Button(
+    {
+      children,
+      iconLeft,
+      iconRight,
+      disabled,
+      type,
+      buttonType = "primary",
+      isLoading = false,
+      className,
+      ...rest
+    },
+    ref
+  ) {
+    const formContext = useFormContext();
+    let _disabled = disabled;
+    let _isLoading = isLoading;
+    if (formContext !== null && type === "submit") {
+      const {
+        formState: { isSubmitting, errors },
+      } = formContext;
+      // Can't use the built-in isValid on formState because that only updates when the mode is set to onChange/onBlur/whatever
+      const isValid = Object.keys(errors).length === 0;
+      _disabled = disabled || isSubmitting || !isValid;
+      _isLoading = isSubmitting;
+    }
+    const spinnerOnLeft = _isLoading && iconLeft;
+    const spinnerOnRight = _isLoading && !spinnerOnLeft;
     return (
-      <Button
-        iconRight={icon}
+      <button
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        ref={ref as any}
         className={clsx(
-          size === "sm"
-            ? "!p-0.5"
-            : size === "md"
-            ? "!p-1.5"
-            : size === "lg"
-            ? "!p-2.5"
-            : size === "xl"
-            ? "!p-[1.125rem]"
-            : null,
+          "rounded-[100px]",
+          "flex items-center",
+          children && "px-6",
+          {
+            "bg-accent-1 text-green-90 hover:bg-accent-1-shade disabled:opacity-40 disabled:hover:bg-accent-1":
+              buttonType == "primary",
+            "border border-accent-1 bg-white text-accent-1 hover:bg-accent-1/10 disabled:border-dark-80 disabled:text-dark-80 disabled:hover:bg-white":
+              buttonType == "secondary",
+            "bg-transparent text-white hover:bg-dark-80 disabled:text-dark-80 disabled:hover:bg-transparent":
+              buttonType == "tertiary",
```

Comment:
I think you gonna need to fix those classes with the dark theme, when I try secondary type, the hover class is wrong.

Reply:
Sorry..I think it's fixed now.